src/ark/utils/data_utils.py

```diff
@@ -537,9 +539,16 @@ def generate_pixel_cluster_mask(fov, base_dir, tiff_dir, chan_file_path,
     # convert to 1D indexing
     coordinates = x_coords * img_data.shape[1] + y_coords
 
-    # get the cooresponding cluster labels for each pixel
+    # get the corresponding cluster labels for each pixel
     cluster_labels = list(fov_data[pixel_cluster_col])
 
+    # relabel clusters with sequential integers (cluster_id)
+    unique_clusters = list(np.unique(cluster_labels))  # returns sorted meta cluster numbers
+    cluster_ids = list(range(1, len(unique_clusters) + 1))
+    id_mapping = {meta_cluster: cluster_id
+                  for meta_cluster, cluster_id in zip(unique_clusters, cluster_ids)}
+    cluster_labels = [id_mapping[label] for label in cluster_labels]
+
```

Comment:
When I search for this function `generate_pixel_cluster_mask` in Github, I can't tell where in the pipeline it's used (search results only return the function definition and the test file), but I imagine that this implementation could cause some downstream problems if not every FOV contains every cluster. For example, if FOV1 contains clusters 1,3,5 - then those will be mapped to 1,2,3. However, if FOV2 contains clusters 1,2,5 - then those will also be mapped to 1,2,3. This could be confusing downstream because it seems like FOV1 and FOV2 contain the same clusters when they don't.

Reply:
It's used in the same data_utils files in `generate_and_save_pixel_cluster_masks()`.
But yes you're right this could definitely cause issues. I can instead pass the mapping saved to cluster_id_name_path in the wrapper function since that is generated directly from the mapping file which contains every cluster!

Reply:
Oh duh I'm dumb lol